cmd/backup/main.go

```diff
@@ -330,10 +332,28 @@ func (s *script) stopContainers() (func() error, error) {
 // saves it to disk.
 func (s *script) takeBackup() error {
 	s.file = timeutil.Strftime(&s.start, s.file)
-	if err := targz.Compress(s.c.BackupSources, s.file); err != nil {
+	backupSources := s.c.BackupSources
+	if s.c.BackupFromSnapshot {
+		backupCopiedSource := filepath.Join("/tmp", s.c.BackupSources)
+		backupSources = backupCopiedSource
+		s.logger.Infof("Created snapshot of `%s` at `%s`.", s.c.BackupSources, backupCopiedSource)
+		// copy before compressing guard against a situation where backup folder's content are still growing.
+		if err := copy.Copy(s.c.BackupSources, backupCopiedSource, copy.Options{ PreserveTimes: true }); err != nil {
+			return fmt.Errorf("takeBackup: error creating snapshot on backup folder: %w", err)
+		}
+		defer func() {
```

Comment:
so honestly I have never written go before and idk if this is what you are thinking, I tested it and it seems ok. Guess I really need to learn more about go if I am going to continue to make pr here

Reply:
That's exactly how defer works :) no need to worry.